js/pro/poloniex.js

```diff
@@ -0,0 +1,886 @@
+         */
+        const name = 'ticker';
+        return await this.subscribe (name, 'public', [ symbol ], params);
+    }
+
+    async watchTickers (symbols, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://docs.poloniex.com/#public-channels-market-data-ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const name = 'ticker';
+        if (symbols === undefined) {
+            symbols = [ 'all' ];
+        }
+        return await this.subscribe (name, 'public', symbols, params);
+    }
+
+    async watchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @see https://docs.poloniex.com/#public-channels-market-data-trades
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const name = 'trades';
+        const trades = await this.subscribe (name, 'public', [ symbol ], params);
+        if (this.newUpdates) {
+            limit = trades.getLimit (symbol, limit);
+        }
+        return this.filterBySinceLimit (trades, since, limit, 'timestamp', true);
+    }
+
+    async watchOrderBook (symbol, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchOrderBook
+         * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://docs.poloniex.com/#public-channels-market-data-book-level-2
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int|undefined} limit not used by poloniex watchOrderBook
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-book-structure} indexed by market symbols
+         */
+        const name = 'book_lv2';
```

Comment:
Since we have two available book channels (book and book_lv2) this should be configurable through options + params.

Like this:
```Javascript
        let name = undefined
        [ name, params ] = this.handleOptionAndParams (params, 'watchOrderBook', 'name', ''book_lv2');
```